File: src/main/java/com/github/zametki/model/ZType.java
package com.github.zametki.model;

import com.github.mjdbc.type.DbInt;

public enum ZType implements DbInt {
    PLAIN_TEXT(0);

    public final int id;

    ZType(int id) {
        this.id = id;
    }


    @Override
    public int getDbValue() {
        return 0;
    }

    public static ZType fromDbValue(int id) {
        if (id != 0) {
            throw new IllegalArgumentException("Unsupported content type: " + id);
        }
        return PLAIN_TEXT;
    }
}

File: src/main/java/com/github/zametki/model/ZType.kt
package com.github.zametki.model

import com.github.mjdbc.type.DbInt

enum class ZType(val id: Int) : DbInt {
    PLAIN_TEXT(0);


    override fun getDbValue() = 0

    companion object {

        fun fromDbValue(id: Int): ZType {
            if (id != 0) {
                throw IllegalArgumentException("Unsupported content type: " + id)
            }
            return PLAIN_TEXT
        }
    }
}
